filesystems/filesystems.tex

```diff
@@ -1159,82 +1181,100 @@ \subsection{How big is our file?}
 We would once again fetch this number from the inode, go to the root of the data blocks, go to the $5$th data block and perform our write at the $2$KiB offset from this block because we skipped the first four kibibytes of the file in block 7.
-We perform our write and go on our merry way
+We perform our write and go on our merry way.
 
-Some questions for you. How would you consider performing a write that would go across data block boundaries? How would you consider performing a write whose write after adding the offset would extend the length of the file?  How would you consider performing a write where the offset is greater than the length of the original file?
+Some questions for you:
 
-Performing a write to a directory meaning that an inode needs to be added to a directory.
+\begin {itemize}
+    \item How would you consider performing a write that would go across data block boundaries?
+    \item How would you consider performing a write whose write after adding the offset would extend the length of the file?\item How would you consider performing a write where the offset is greater than the length of the original file?
+\end{itemize}
+
+\subsubsection{Writing to directories}
+Performing a write to a directory implies that an inode needs to be added to a directory.
 If we pretend that the example above is a directory.
 We know that we will be adding at most one directory entry at a time.
 Meaning that we have to have enough space for one directory entry in our data blocks.
 Luckily the last data block that we have has enough free space.
 This means we just need to find the number of the last data block as we did above, go to where the data ends, and write one directory entry.
 Don't forget to update the size of the directory so that the next creation doesn't overwrite your file!
 
-Some more questions. How would you consider performing a write when the last data block is already full? How about when all the direct blocks have just been filled up and the inode doesn't have an indirect block? What about when the first indirect entry (\#4) is full?
-These are all edge cases you have to think about for a filesystem that really get you
+Some more questions: 
+
+\begin{itemize}
+    \item How would you consider performing a write when the last data block is already full?
+    \item How about when all the direct blocks have just been filled up and the inode doesn't have an indirect block?
+    \item What about when the first indirect entry (\#4) is full?
+\end{itemize}
+
+These are edge cases you have to think about for a filesystem that really get you.
 
 \subsection{Adding Deletes}
 
-Although this isn't part of the lab originally.
-If you were to ask what happens when a file gets deleted, it's actually pretty simple.
+Although this wasn't part of the CS 241 assignment originally, if you were to ask what happens when a file gets deleted, it is actually quite simple.
```

Comment:
We could just get rid of this sentence. The less tie in we have with our assignments, the better.